File: apps/bff/src/modules/pub/router.ts
import { publicProcedure, router } from '@bff/trpc/init';
import type { AppConfigResponse } from '@tooday/shared';

/** 공유 캐시 대상 — 유저별로 달라지는 데이터 금지 (ctx.user 참조 금지) */
export const pubRouter = router({
  appConfig: publicProcedure.query(
    (): AppConfigResponse => ({
      version: '0.0.0',
      minSupportedAppVersion: '0.0.0',
      features: {
        projects: true,
        timeline: true,
      },
    }),
  ),
});

File: rust/crates/bff/src/modules/pub_/router.rs
use tooday_shared::{AppConfigResponse, AppFeatures};

use crate::trpc::init::{AuthLevel, Procedure, ProcedureKind};

/// 공유 캐시 대상 — 유저별로 달라지는 데이터 금지 (컨텍스트의 userId 참조 금지)
pub const PROCEDURES: &[Procedure] =
    &[Procedure { name: "appConfig", kind: ProcedureKind::Query, auth: AuthLevel::Public }];

pub fn app_config() -> AppConfigResponse {
    AppConfigResponse {
        version: "0.0.0".into(),
        min_supported_app_version: "0.0.0".into(),
        features: AppFeatures { projects: true, timeline: true },
    }
}
